gradle-plugin/src/main/java/com/contrastsecurity/gradle/plugin/ContrastVerifyTestTask.java

```diff
@@ -1,15 +1,229 @@
+    }
+
+    if (traces != null && traces.getCount() > 0) {
+      logger.info(traces.getCount() + " new vulnerability(s) were found.");
+
+      // Write TeamServer results to a file in the build directory
+      try (BufferedWriter writer =
+          new BufferedWriter(new FileWriter(getTraceResults().get().getAsFile(), true))) {
+        for (Trace trace : traces.getTraces()) {
+          final String traceReport = generateTraceReport(trace);
+          writer.write(traceReport);
+          writer.newLine();
+          System.out.println(traceReport);
+        }
+      } catch (final IOException e) {
+        logger.debug("Could not write Contrast Findings to build directory: " + e);
+      }
+
+      throw new GradleException(
+          "Your application is vulnerable. Please see the above report for new vulnerabilities.");
+    } else {
+      logger.info("No new vulnerabilities were found.");
+    }
+
+    logger.info("Finished verifying your application.");
+  }
+
+  /**
+   * Retrieves the server id by server name
+   *
+   * @param sdk Contrast SDK object
+   * @param applicationId application id to filter on
+   * @return List<Long> id of the servers
+   */
+  private List<String> getServerId(
+      final ContrastSDK sdk,
+      final String applicationId,
+      final String orgUuid,
+      final String serverName) {
+    ServerFilterForm serverFilterForm = new ServerFilterForm();
+    serverFilterForm.setApplicationIds(Collections.singletonList(applicationId));
+
+    Servers servers;
+    List<String> serverIds;
+
+    try {
+      serverFilterForm.setQ(URLEncoder.encode(serverName, "UTF-8"));
+      servers = sdk.getServersWithFilter(orgUuid, serverFilterForm);
+    } catch (IOException e) {
+      throw new GradleException("Unable to retrieve the servers.", e);
+    } catch (UnauthorizedException e) {
+      throw new GradleException("Unable to connect to Contrast.", e);
+    }
+
+    if (!servers.getServers().isEmpty()) {
+      serverIds = new ArrayList<String>();
+      for (Server server : servers.getServers()) {
+        serverIds.add(String.valueOf(server.getServerId()));
+      }
+    } else {
+      throw new GradleException(
+          "\n\nServer with name '"
+              + serverName
+              + "' not found. Make sure this server name appears in Contrast under the 'Servers' tab.\n");
+    }
+
+    return serverIds;
+  }
+
+  /** Retrieves appId from TeamServer using the OrgUuid and Appname */
+  private String getApplicationId(
+      final ContrastConfigurationExtension config, final String orgUuid) {
+    Applications applications;
+
+    if (!config.getAppName().isPresent()) {
+      throw new GradleException("AppName or Organization UUID not set");
+    }
+    try {
+      applications = sdk.getApplications(orgUuid);
+    } catch (IOException e) {
+      throw new GradleException("Unable to retrieve the applications: " + e);
```

Comment:
Don't do `+ e` for these exceptions. We want to provide `e` as the cause so it'll show up when users do `--stacktrace`.

```
throw new GradleException("Unable to retrieve the applications: ", e);
```